yorkie/backend/sync/etcd/client.go

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2021 The Yorkie Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package etcd
+
+import (
+	"context"
+	"time"
+
+	"go.etcd.io/etcd/clientv3"
+	"go.etcd.io/etcd/clientv3/concurrency"
+
+	"github.com/yorkie-team/yorkie/pkg/log"
+	"github.com/yorkie-team/yorkie/yorkie/backend/sync"
+)
+
+// DefaultLockLeaseTimeSec is the default lease time of lock.
+const DefaultLockLeaseTimeSec = 30
+
+// Config is the configuration for creating a Client instance.
+type Config struct {
+	Endpoints      []string      `json:"Endpoints"`
+	DialTimeoutSec time.Duration `json:"DialTimeoutSec"`
+	Username       string        `json:"Username"`
+	Password       string        `json:"Password"`
+
+	LockLeaseTimeSec int `json:"LockLeaseTimeSec"`
+}
+
+// Client is a client that connects to ETCD.
+type Client struct {
+	config *Config
+	client *clientv3.Client
+}
+
+// NewClient creates a new instance of Client.
+func NewClient(conf *Config) *Client {
```

Comment:
I seem to be able to change this method to a private method.
Currently, it is called in the `TestETCDStress` test of `etcd_stress_test.go`, but it seems that it can be replaced with `Dial(conf *Config)`.